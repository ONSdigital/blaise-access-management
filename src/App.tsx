import React, { ReactElement, useEffect, useState } from "react";
import { isDevEnv } from "./Functions";
import { Switch, Route, useLocation } from "react-router-dom";
import Users from "./pages/users/Users";
import NewUserComponent from "./pages/users/NewUser";
import ChangePassword from "./pages/users/ChangePassword";
import DeleteUser from "./pages/users/DeleteUser";
import { NotProductionWarning, Footer, Header, BetaBanner, ErrorBoundary, DefaultErrorBoundary, ONSLoadingPanel } from "blaise-design-system-react-components";
import Roles from "./pages/roles/Roles";
import BulkUserUpload from "./pages/users/BulkUserUpload/BulkUserUpload";
import Home from "./pages/Home";
import { LoginForm, AuthManager } from "blaise-login-react-client";
import { User } from "blaise-api-node-client";
import { getCurrentUser } from "blaise-login-react-client";

interface window extends Window {
    VM_EXTERNAL_CLIENT_URL: string
    CATI_DASHBOARD_URL: string
}

const divStyle = {
    minHeight: "calc(67vh)"
};

function App(): ReactElement {
    const authManager = new AuthManager();
    const [externalCATIUrl, setExternalCATIUrl] = useState<string>("/Blaise");
    const location = useLocation();
    const [loaded, setLoaded] = useState(false);
    const [loggedIn, setLoggedIn] = useState(false);
    const [currentUser, setCurrentUser] = useState<User>();

    useEffect(() => {
        console.log(location);
        authManager.loggedIn().then(async (isLoggedIn: boolean) => {
            setLoggedIn(isLoggedIn);
            if (loggedIn) {
                getCurrentUser(authManager).then((user: User) => {
                    setCurrentUser(user);
                }).catch(() => {
                    console.log("IDK what is going on");
                });
            }
            setLoaded(true);
        });
    });


    useEffect(function retrieveVariables() {
        setExternalCATIUrl(isDevEnv() ?
            process.env.REACT_APP_CATI_DASHBOARD_URL || externalCATIUrl : (window as unknown as window).CATI_DASHBOARD_URL);
    }, [externalCATIUrl]);

    function loginPage(): ReactElement {
        if (loaded && loggedIn) {
            return <></>;
        }
        return (
            <div style={divStyle} className="ons-page__container ons-container">
                <LoginForm authManager={authManager} setLoggedIn={setLoggedIn} />
            </div>
        );
    }

    function signOut(): void {
        authManager.clearToken();
        setLoggedIn(false);
    }

    function loading(): ReactElement {
        if (loaded) {
            return <></>;
        }
        return (
            <div style={divStyle} className="ons-page__container ons-container">
                <ONSLoadingPanel />
            </div>
        );
    }

    function app(): ReactElement {
        if (loaded && loggedIn) {
            return (
                <>
                    {/* <NavigationLinks /> */}
                    <div style={divStyle} className="ons-page__container ons-container">
                        <DefaultErrorBoundary>
                            <Switch>
                                <Route path={"/users/upload"}>
                                    <BulkUserUpload />
                                </Route>
                                <Route path={"/users/changepassword/:user"}>
                                    <ChangePassword />
                                </Route>
                                <Route path={"/users/delete/:user"}>
                                    <DeleteUser />
                                </Route>
                                <Route path={"/users/new"}>
                                    <NewUserComponent />
                                </Route>
                                <Route path={"/roles"}>
                                    <ErrorBoundary errorMessageText={"Unable to load role table correctly."}>
                                        <Roles />
                                    </ErrorBoundary>
                                </Route>
                                <Route path="/users">
                                    <ErrorBoundary errorMessageText={"Unable to load user table correctly."}>
                                        <Users currentUser={currentUser}
                                            externalCATIUrl={externalCATIUrl} />
                                    </ErrorBoundary>
                                </Route>
                                <Route path="/">
                                    <ErrorBoundary errorMessageText={"Unable to load user table correctly."}>
                                        <Home user={currentUser} />
                                    </ErrorBoundary>
                                </Route>
                            </Switch>
                        </DefaultErrorBoundary>
                    </div>
                </>
            );
        }
        return <></>;
    }

    return (
        <>
            {
                (window.location.hostname.includes("dev")) && <NotProductionWarning />
            }
            <BetaBanner />
            <Header 
                title={"Blaise User Management"} 
                signOutButton={loggedIn} 
                noSave={true} 
                signOutFunction={signOut} 
                navigationLinks={
                    [   
                        {
                            id: "home-link",
                            label: "Home",
                            endpoint: "/"
                        }, 
                        {
                            id: "users-link",
                            label: "Manage users",
                            endpoint: "/users"
                        }, 
                        {
                            id: "roles-link",
                            label: "Manage roles",
                            endpoint: "/roles"
                        },
                    ]
                }     
            />
            {loading()}
            {loginPage()}
            {app()}
            <Footer />
        </>
    );
}

export default App;
