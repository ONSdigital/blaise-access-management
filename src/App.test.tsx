import React from "react";
import {render, waitFor, cleanup, screen} from "@testing-library/react";
import App from "./App";
import "@testing-library/jest-dom";
import {act} from "react-dom/test-utils";
import {createMemoryHistory} from "history";
import {Router} from "react-router";
import {AuthManager} from "blaise-login-react-client";
import * as loginReactClient from "blaise-login-react-client";

jest.mock("blaise-login-react-client");

AuthManager.prototype.loggedIn = jest.fn().mockImplementation(() => {
    return Promise.resolve(true);
});

jest.spyOn(loginReactClient, "getCurrentUser").mockReturnValue(
    Promise.resolve({
        defaultServerPark: "gusty",
        name: "Blaise User",
        role: "DST",
        serverParks: ["gusty"]
    })
);

describe("React homepage", () => {
    it("the homepage matches Snapshot", async () => {
        const history = createMemoryHistory();
        const wrapper = render(
            <Router history={history}>
                <App />
            </Router>
        );

        await act(async () => {
            await new Promise(process.nextTick);
        });

        await waitFor(() => {
            expect(wrapper).toMatchSnapshot();
        });

    });

    it("should render correctly", async () => {
        const history = createMemoryHistory();
        render(
            <Router history={history}>
                <App />
            </Router>
        );
        expect(screen.getByText(/Blaise Access Management/i)).toBeDefined();

        await act(async () => {
            await new Promise(process.nextTick);
        });

        await waitFor(() => {
            expect(screen.getByText(/Blaise Access Management/i)).toBeDefined();
            expect(screen.queryAllByText(/Manage users/i)).toBeDefined();
            expect(screen.queryAllByText(/Manage roles/i)).toBeDefined();
            expect(screen.queryByText(/Loading/i)).not.toBeInTheDocument();
        });
    });

    afterAll(() => {
        jest.clearAllMocks();
        cleanup();
    });
});
