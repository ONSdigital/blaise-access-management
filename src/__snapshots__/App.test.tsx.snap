// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React homepage view questionnaires page matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="phase-banner"
      >
        <div
          class="container "
        >
          <div
            class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-no-grow"
            >
              <h3
                class="phase-banner__badge"
              >
                BETA
              </h3>
            </div>
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <p
                class="phase-banner__desc u-fs-s u-mb-no"
              >
                This is a new service – your 
                <a
                  href="https://ons.service-now.com/"
                >
                  feedback
                </a>
                 will help us improve it.
              </p>
            </div>
          </div>
        </div>
      </div>
      <header
        class="header header--internal"
      >
        <div
          class="header__top"
          role="banner"
        >
          <div
            class="container"
          >
            <div
              class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
            >
              <div
                class="grid__col col-auto"
              >
                <a
                  class="header__logo-link"
                  href="/"
                >
                  <picture>
                    <img
                      alt="Office for National Statistics logo"
                      class="header__logo"
                      src="https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg"
                    />
                  </picture>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="header__main"
        >
          <div
            class="container"
          >
            <div
              class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
            >
              <div
                class="grid__col col-auto u-flex-shrink"
              >
                <div
                  class="header__title"
                >
                  Blaise User Management
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="page__container container"
        style="min-height: calc(67vh);"
      >
        <main
          class="page__main"
          id="main-content"
        >
          <h1>
            Interviewing
          </h1>
          <p>
            This page provides information on active questionnaires with corresponding links that redirect to specific areas of CATI dashboard.
          </p>
          <p>
            Please note, the table containing information on active questionnaires may take a few seconds to load.
          </p>
          <p
            class="u-mt-m"
          >
            <a
              class="external-link"
              id="cati-dashboard"
              rel="noopener noreferrer"
              target="_blank"
            >
              Link to CATI dashboard
              <svg
                class="svg-icon"
                viewBox="0 0 12 12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z"
                  transform="translate(-2 -1.99)"
                />
                <path
                  d="M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z"
                  transform="translate(-2 -1.99)"
                />
              </svg>
            </a>
          </p>
          <p>
            <a
              href="/"
              id="return-to-survey-list"
            >
              Return to survey list
            </a>
          </p>
          <h3>
            Active questionnaires
          </h3>
          <table
            class="table "
            id="instrument-table"
          >
            <thead
              class="table__head u-mt-m"
            >
              <tr
                class="table__row"
              >
                <th
                  class="table__header "
                  scope="col"
                >
                  <span>
                    Questionnaire
                  </span>
                </th>
                <th
                  class="table__header "
                  scope="col"
                >
                  <span>
                    Field period
                  </span>
                </th>
                <th
                  class="table__header "
                  scope="col"
                >
                  <span>
                    Link to interview
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              class="table__body"
            >
              <tr
                class="table__row"
                data-testid="instrument-table-row"
              >
                <td
                  class="table__cell "
                >
                  OPN2007T
                </td>
                <td
                  class="table__cell "
                >
                  July 2020
                </td>
                <td
                  class="table__cell "
                >
                  <a
                    aria-label="Launch interview for instrument OPN2007T July 2020"
                    class="external-link"
                    href="https://external-web-url/OPN2007T?LayoutSet=CATI-Interviewer_Large"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Interview
                    <svg
                      class="svg-icon"
                      viewBox="0 0 12 12"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z"
                        transform="translate(-2 -1.99)"
                      />
                      <path
                        d="M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z"
                        transform="translate(-2 -1.99)"
                      />
                    </svg>
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
        </main>
      </div>
      <footer
        class="footer"
      >
        <div
          class="footer--body"
          data-analytics="footer"
        >
          <div
            class="container "
          >
            <div
              class="grid"
            >
              <div
                class="u-mb-m"
              >
                <dl
                  aria-label="Blaise Server Manager link to telephone server park"
                  class="metadata metadata__list grid grid--gutterless u-cf u-mb-l"
                  title="Blaise Server Manager link to telephone server park"
                >
                  <dt
                    class="metadata__term grid__col col-6@m"
                  >
                     Telephone server park for Blaise Server Manager:
                  </dt>
                  <dd
                    class="metadata__value grid__col col-6@m"
                  />
                </dl>
                <hr
                  class="footer__hr"
                />
              </div>
            </div>
            <div
              class="grid grid--flex grid--between"
            >
              <div
                class="grid__col"
              >
                <div
                  class="u-mt-m u-mb-s"
                >
                  <img
                    alt="Office for National Statistics"
                    src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="phase-banner"
    >
      <div
        class="container "
      >
        <div
          class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
        >
          <div
            class="grid__col col-auto u-flex-no-grow"
          >
            <h3
              class="phase-banner__badge"
            >
              BETA
            </h3>
          </div>
          <div
            class="grid__col col-auto u-flex-shrink"
          >
            <p
              class="phase-banner__desc u-fs-s u-mb-no"
            >
              This is a new service – your 
              <a
                href="https://ons.service-now.com/"
              >
                feedback
              </a>
               will help us improve it.
            </p>
          </div>
        </div>
      </div>
    </div>
    <header
      class="header header--internal"
    >
      <div
        class="header__top"
        role="banner"
      >
        <div
          class="container"
        >
          <div
            class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
          >
            <div
              class="grid__col col-auto"
            >
              <a
                class="header__logo-link"
                href="/"
              >
                <picture>
                  <img
                    alt="Office for National Statistics logo"
                    class="header__logo"
                    src="https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg"
                  />
                </picture>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="header__main"
      >
        <div
          class="container"
        >
          <div
            class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <div
                class="header__title"
              >
                Blaise User Management
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div
      class="page__container container"
      style="min-height: calc(67vh);"
    >
      <main
        class="page__main"
        id="main-content"
      >
        <h1>
          Interviewing
        </h1>
        <p>
          This page provides information on active questionnaires with corresponding links that redirect to specific areas of CATI dashboard.
        </p>
        <p>
          Please note, the table containing information on active questionnaires may take a few seconds to load.
        </p>
        <p
          class="u-mt-m"
        >
          <a
            class="external-link"
            id="cati-dashboard"
            rel="noopener noreferrer"
            target="_blank"
          >
            Link to CATI dashboard
            <svg
              class="svg-icon"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z"
                transform="translate(-2 -1.99)"
              />
              <path
                d="M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z"
                transform="translate(-2 -1.99)"
              />
            </svg>
          </a>
        </p>
        <p>
          <a
            href="/"
            id="return-to-survey-list"
          >
            Return to survey list
          </a>
        </p>
        <h3>
          Active questionnaires
        </h3>
        <table
          class="table "
          id="instrument-table"
        >
          <thead
            class="table__head u-mt-m"
          >
            <tr
              class="table__row"
            >
              <th
                class="table__header "
                scope="col"
              >
                <span>
                  Questionnaire
                </span>
              </th>
              <th
                class="table__header "
                scope="col"
              >
                <span>
                  Field period
                </span>
              </th>
              <th
                class="table__header "
                scope="col"
              >
                <span>
                  Link to interview
                </span>
              </th>
            </tr>
          </thead>
          <tbody
            class="table__body"
          >
            <tr
              class="table__row"
              data-testid="instrument-table-row"
            >
              <td
                class="table__cell "
              >
                OPN2007T
              </td>
              <td
                class="table__cell "
              >
                July 2020
              </td>
              <td
                class="table__cell "
              >
                <a
                  aria-label="Launch interview for instrument OPN2007T July 2020"
                  class="external-link"
                  href="https://external-web-url/OPN2007T?LayoutSet=CATI-Interviewer_Large"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Interview
                  <svg
                    class="svg-icon"
                    viewBox="0 0 12 12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.5,9H13a.5.5,0,0,0-.5.5v3h-9v-9h3A.5.5,0,0,0,7,3V2.5A.5.5,0,0,0,6.5,2h-4a.5.5,0,0,0-.5.5v11a.5.5,0,0,0,.5.5h11a.5.5,0,0,0,.5-.5v-4A.5.5,0,0,0,13.5,9Z"
                      transform="translate(-2 -1.99)"
                    />
                    <path
                      d="M8.83,7.88a.51.51,0,0,0,.71,0l2.31-2.32,1.28,1.28A.51.51,0,0,0,14,6.49v-4a.52.52,0,0,0-.5-.5h-4A.51.51,0,0,0,9,2.52a.58.58,0,0,0,.14.33l1.28,1.28L8.12,6.46a.51.51,0,0,0,0,.71Z"
                      transform="translate(-2 -1.99)"
                    />
                  </svg>
                </a>
              </td>
            </tr>
          </tbody>
        </table>
      </main>
    </div>
    <footer
      class="footer"
    >
      <div
        class="footer--body"
        data-analytics="footer"
      >
        <div
          class="container "
        >
          <div
            class="grid"
          >
            <div
              class="u-mb-m"
            >
              <dl
                aria-label="Blaise Server Manager link to telephone server park"
                class="metadata metadata__list grid grid--gutterless u-cf u-mb-l"
                title="Blaise Server Manager link to telephone server park"
              >
                <dt
                  class="metadata__term grid__col col-6@m"
                >
                   Telephone server park for Blaise Server Manager:
                </dt>
                <dd
                  class="metadata__value grid__col col-6@m"
                />
              </dl>
              <hr
                class="footer__hr"
              />
            </div>
          </div>
          <div
            class="grid grid--flex grid--between"
          >
            <div
              class="grid__col"
            >
              <div
                class="u-mt-m u-mb-s"
              >
                <img
                  alt="Office for National Statistics"
                  src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`React homepage view surveys page matches Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="phase-banner"
      >
        <div
          class="container "
        >
          <div
            class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-no-grow"
            >
              <h3
                class="phase-banner__badge"
              >
                BETA
              </h3>
            </div>
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <p
                class="phase-banner__desc u-fs-s u-mb-no"
              >
                This is a new service – your 
                <a
                  href="https://ons.service-now.com/"
                >
                  feedback
                </a>
                 will help us improve it.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="census-warning"
        style="background: rgb(208, 2, 27); color: rgb(208, 2, 27);"
      >
        <div
          class="container"
        >
          <div
            class="panel panel--warn-branded panel--no-title"
            style="background: rgb(208, 2, 27);"
          >
            <span
              aria-hidden="true"
              class="panel__icon"
              style="color: rgb(208, 2, 27);"
            >
              !
            </span>
            <div
              class="panel__body"
            >
              <p
                class="u-mb-no"
              >
                This environment is not a production environment. Do not upload any live data to this service.
              </p>
            </div>
          </div>
        </div>
      </div>
      <header
        class="header header--internal"
      >
        <div
          class="header__top"
          role="banner"
        >
          <div
            class="container"
          >
            <div
              class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
            >
              <div
                class="grid__col col-auto"
              >
                <a
                  class="header__logo-link"
                  href="/"
                >
                  <picture>
                    <img
                      alt="Office for National Statistics logo"
                      class="header__logo"
                      src="https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg"
                    />
                  </picture>
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="header__main"
        >
          <div
            class="container"
          >
            <div
              class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
            >
              <div
                class="grid__col col-auto u-flex-shrink"
              >
                <div
                  class="header__title"
                >
                  Blaise User Management
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div
        class="page__container container"
        style="min-height: calc(67vh);"
      >
        <main
          class="page__main"
          id="main-content"
        >
          <div
            class="panel panel--info panel--simple "
            hidden=""
          >
            <div
              class="panel__body"
            >
              <p />
            </div>
          </div>
          <h1>
            Sign in
          </h1>
          <p />
          <form>
            <p
              class="field"
            >
              <label
                class="label"
              >
                Username
              </label>
              <input
                class="input input--text input-type__input "
                data-testid="text-input"
                type="text"
                value=""
              />
            </p>
            <p
              class="field"
            >
              <label
                class="label"
                for="password"
              >
                Password
              </label>
              <span
                class="checkbox checkbox--toggle"
              >
                <input
                  autocomplete="new-password"
                  class="checkbox__input"
                  id="password-toggle"
                  name="show-password"
                  type="checkbox"
                />
                <label
                  class="checkbox__label "
                  for="password-toggle"
                  id="password-toggle-label"
                >
                  Show password
                </label>
              </span>
              <input
                class="input input--text input-type__input u-mt-xs"
                data-testid="login-password-input"
                id="password"
                type="password"
                value=""
              />
            </p>
            <button
              class="btn "
              data-testid="button"
              type="button"
            >
              <span
                class="btn__inner"
              >
                Sign in
              </span>
            </button>
          </form>
        </main>
      </div>
      <footer
        class="footer"
      >
        <div
          class="footer__body"
          data-analytics="footer"
        >
          <div
            class="container "
          >
            <div
              class="grid grid--flex grid--between"
            >
              <div
                class="grid__col"
              >
                <div
                  class="u-mt-m u-mb-s"
                >
                  <img
                    alt="Office for National Statistics"
                    src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <div
      class="phase-banner"
    >
      <div
        class="container "
      >
        <div
          class="grid grid--flex grid--gutterless grid--vertical-center grid--no-wrap"
        >
          <div
            class="grid__col col-auto u-flex-no-grow"
          >
            <h3
              class="phase-banner__badge"
            >
              BETA
            </h3>
          </div>
          <div
            class="grid__col col-auto u-flex-shrink"
          >
            <p
              class="phase-banner__desc u-fs-s u-mb-no"
            >
              This is a new service – your 
              <a
                href="https://ons.service-now.com/"
              >
                feedback
              </a>
               will help us improve it.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="census-warning"
      style="background: rgb(208, 2, 27); color: rgb(208, 2, 27);"
    >
      <div
        class="container"
      >
        <div
          class="panel panel--warn-branded panel--no-title"
          style="background: rgb(208, 2, 27);"
        >
          <span
            aria-hidden="true"
            class="panel__icon"
            style="color: rgb(208, 2, 27);"
          >
            !
          </span>
          <div
            class="panel__body"
          >
            <p
              class="u-mb-no"
            >
              This environment is not a production environment. Do not upload any live data to this service.
            </p>
          </div>
        </div>
      </div>
    </div>
    <header
      class="header header--internal"
    >
      <div
        class="header__top"
        role="banner"
      >
        <div
          class="container"
        >
          <div
            class="header__grid-top grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap "
          >
            <div
              class="grid__col col-auto"
            >
              <a
                class="header__logo-link"
                href="/"
              >
                <picture>
                  <img
                    alt="Office for National Statistics logo"
                    class="header__logo"
                    src="https://ons-design-system.netlify.app/img/ons-logo-neg-en.svg"
                  />
                </picture>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        class="header__main"
      >
        <div
          class="container"
        >
          <div
            class="grid grid--gutterless grid--flex grid--between grid--vertical-center grid--no-wrap"
          >
            <div
              class="grid__col col-auto u-flex-shrink"
            >
              <div
                class="header__title"
              >
                Blaise User Management
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div
      class="page__container container"
      style="min-height: calc(67vh);"
    >
      <main
        class="page__main"
        id="main-content"
      >
        <div
          class="panel panel--info panel--simple "
          hidden=""
        >
          <div
            class="panel__body"
          >
            <p />
          </div>
        </div>
        <h1>
          Sign in
        </h1>
        <p />
        <form>
          <p
            class="field"
          >
            <label
              class="label"
            >
              Username
            </label>
            <input
              class="input input--text input-type__input "
              data-testid="text-input"
              type="text"
              value=""
            />
          </p>
          <p
            class="field"
          >
            <label
              class="label"
              for="password"
            >
              Password
            </label>
            <span
              class="checkbox checkbox--toggle"
            >
              <input
                autocomplete="new-password"
                class="checkbox__input"
                id="password-toggle"
                name="show-password"
                type="checkbox"
              />
              <label
                class="checkbox__label "
                for="password-toggle"
                id="password-toggle-label"
              >
                Show password
              </label>
            </span>
            <input
              class="input input--text input-type__input u-mt-xs"
              data-testid="login-password-input"
              id="password"
              type="password"
              value=""
            />
          </p>
          <button
            class="btn "
            data-testid="button"
            type="button"
          >
            <span
              class="btn__inner"
            >
              Sign in
            </span>
          </button>
        </form>
      </main>
    </div>
    <footer
      class="footer"
    >
      <div
        class="footer__body"
        data-analytics="footer"
      >
        <div
          class="container "
        >
          <div
            class="grid grid--flex grid--between"
          >
            <div
              class="grid__col"
            >
              <div
                class="u-mt-m u-mb-s"
              >
                <img
                  alt="Office for National Statistics"
                  src="https://ons-design-system.netlify.app/img/ons-logo-black-en.svg"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
